### PR TITLE
Implement a pattern library for SilverStripe4 react components

### PR DESCRIPTION
**Pull request by flamerohr**
Requries https://github.com/silverstripe/webpack-config/pull/10 to be merged (and then package version bumped)

screen shot 2017-10-13 at 4 59 52 pm

Basically, you implement stories with the filename `<Component>-story.js` format, follow the standard template which storybook recommends and run `yarn && yarn storybook` in `vendor/silverstripe/admin`

### CODE REVIEW

File: .storybook/addons.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+import '@storybook/addon-actions/register';
+import '@storybook/addon-links/register';
```
[the remark this answers is not in the document]
Comment (flamerohr) [reply .storybook/addons.js]: Good point, will remove as we don't have plans for that yet

File: webpack-vars.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const Path = require('path');
+
+const PATHS = {
```
[the remark this answers is not in the document]
Comment (flamerohr) [reply webpack-vars.js]: I'd prefer not... it's similar across all "core" modules, but other modules which use `webpack-config` shouldn't have the same structure enforced

File: .storybook/config.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { configure } from '@storybook/react';
+
+// dynamically load any `bundle.scss` or `Component-story.jsx` files
+const modules = [
+  require.context('client/src', true, /(\/bundle\.scss|[A-Za-z]-story\.jsx?)$/),
```
[the remark this answers is not in the document]
Comment (flamerohr) [reply .storybook/config.js]: Can't - this whole `modules` array needs to be statically analysed by webpack (it scans the filesystem and collects the data into the array for us)
it's like how I wanted to move all this into a loop, but for the same reason I can't...

[the remark this answers is not in the document]
Comment (phalkunz) [reply .storybook/config.js]: Ah right!

File: webpack-vars.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (phalkunz) [reply webpack-vars.js]: Forgot about other modules beside core modules

**Comment by flamerohr** [comment]
@phalkunz have rebased, squashed and written some docs which can be found here https://github.com/silverstripe/silverstripe-framework/pull/7657 :)

**Comment by flamerohr** [comment]
DO NOT MERGE

Something broke while rebasing :/

**Pull request merged**